packages/flutter_hooks/lib/src/throttled.dart

```diff
@@ -0,0 +1,42 @@
+part of 'hooks.dart';
+
+/// widget ignore updates accordingly after a specified [duration] duration.
+///
+/// Example:
+/// ```dart
+/// String userInput = ''; // Your input value
+///
+/// // Create a throttle callback
+/// final throttle = useThrottle(duration: const Duration(milliseconds: 500));
+/// // Assume a fetch method fetchData(String query) exists
+/// Button(onPressed: () => throttle(() => fetchData(userInput)));
+/// ```
+void Function(VoidCallback callback) useThrottle({
+  Duration duration = const Duration(milliseconds: 500),
```

Comment:
This value seems arbitrary. I'd rather not have a default value and make the parameter required